app/position/providers/positionprovidersmodel.cpp

```diff
@@ -20,13 +24,40 @@ PositionProvidersModel::PositionProvidersModel( QObject *parent ) : QAbstractLis
     mProviders.push_front( simulated );
   }
 
+  // Keep the names of position providers in sync with names
+  // used in the constructors of the providers...
+
   PositionProvider internal;
   internal.name = tr( "Internal" );
   internal.description = tr( "GPS receiver of this device" );
   internal.providerType = "internal";
   internal.providerId = "devicegps";
 
   mProviders.push_front( internal );
+
+#ifdef ANDROID
+  PositionProvider internalFused;
+  internalFused.name = tr( "Internal (fused)" );
+  if ( AndroidPositionProvider::isFusedAvailable() )
+    internalFused.description = tr( "Using GPS, Wifi and sensors" );
+  else
+    internalFused.description = tr( "Not available (%1)" ).arg( AndroidPositionProvider::fusedErrorString() );
+  internalFused.providerType = "internal";
+  internalFused.providerId = "android_fused";
```

Comment:
How about not even including the fused provider if there is no way to get it working?

Reply:
There are a couple of error states - some of them look like things are broken and can't be fixed easily (SERVICE_MISSING, SERVICE_INVALID), and then there are other that seem fixable (SERVICE_UPDATING, SERVICE_VERSION_UPDATE_REQUIRED, SERVICE_DISABLED) ... play services API even has a bunch of calls to open an error dialog and guide user towards fixing problems (again maybe something to look into later).

I think it makes sense to have it there even if the fused provider is permanently unavailable, so users at least know about it, and can get a better device next time :smile: